src/main/java/com/mixpanel/android/util/RemoteService.java

```diff
@@ -10,7 +10,7 @@
 
 
 public interface RemoteService {
-    boolean isOnline(Context context);
+    boolean isOnline(Context context, OfflineMode offlineMode);
```

Comment:
If we do that then 
* we hide the dependency that `RemoteService` has now with `OfflineMode`.
* this function will be inconsistent with 'performRequest(String endpointUrl, Map<String, Object> params, SSLSocketFactory socketFactory)'. On this case you also can do `MPConfig.getInstance(context).getSSLSocketFactory()`
* any implementation of `RemoteService` will need to know how to get an `OfflineMode` instance as one of his responsibilities
* We need to enforce the necessity to honor client `OfflineMode` via documentation (javadocs maybe) for implementors of `RemoteService`

Do you think that this drawbacks worth the change ?